frontend/app/NewFrontPage.jsx

```diff
@@ -141,6 +143,28 @@ class NewFrontPage extends React.Component {
         this.setState({sortField: newField, sortOrder:newOrder},()=>this.doSort());
     }
 
+    deleteItem(event, item) {
+        console.log('About to delete record for '+item);
+
+        let axiosConfig = {
+            headers: {
+                "Content-Type": "application/json"
+            }
+        }
+
+        const sleep = (milliseconds) => {
+            return new Promise(resolve => setTimeout(resolve, milliseconds))
+        }
+
+        axios.delete("/api/delete/" + item, axiosConfig).then(response=> {
+                this.setState({loading:true});
+                sleep(1000).then(() => {
```

Comment:
this should be in the callback for setState, so that the 1s delay happens _after_ the delete call has returned (it may take >1s to come back).

Something like this:
```
this.setState({loading: true}, ()=>sleep(1000).then(()=>this.refresh()));
```

Reply:
Done.